broker/request-approval.go

```diff
@@ -0,0 +1,91 @@
+package broker
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+)
+
+func RequestApprovalFx(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	var (
+		err    error
+		policy models.Policy
+	)
+
+	log.Println("[RequestApprovalFx] Handler start ----------------------")
+
+	if policy.Status != models.PolicyStatusInitLead {
+		log.Printf("[ProposalFx] cannot request approval for policy with status %s", policy.Status)
+		return "", nil, fmt.Errorf("cannot request approval for policy with status %s", policy.Status)
+	}
+
+	origin = r.Header.Get("Origin")
+	body := lib.ErrorByte(io.ReadAll(r.Body))
+	defer r.Body.Close()
+
+	log.Printf("[RequestApprovalFx] request body: %s", string(body))
+	err = json.Unmarshal(body, &policy)
+	if err != nil {
+		log.Printf("[RequestApprovalFx] error unmarshaling policy: %s", err.Error())
+		return "", nil, err
+	}
+
+	err = requestApproval(&policy)
+	if err != nil {
+		log.Printf("[RequestApprovalFx] error request approval: %s", err.Error())
+		return "", nil, err
+	}
+
+	return "", nil, err
+}
+
+func requestApproval(policy *models.Policy) error {
+	var (
+		err error
+	)
+
+	log.Println("[RequestApproval] start --------------------")
+
+	deniedStatuses := []string{models.PolicyStatusDeleted, models.PolicyStatusRejected}
+
+	if policy.IsReserved && !lib.SliceContains(deniedStatuses, policy.Status) {
+		fmt.Errorf("[RequestApproval] error request approval for policy with uid %s", policy.Uid)
+		return fmt.Errorf("error request approval for policy with uid %s", policy.Uid)
+	}
```

Comment:
Is this correct?
Ex.: policy.IsReserved = true; policy.Status = "NeedsApproval"
=> enters the if and exits as an error
Maybe the control should be:
if !policy.IsReseved || lib.SliceContains(deniedStatuses, policy.Status) {}
=> policy not in reserved => error
=> otherwise if is in one of the denied statuses => error